src/include/sof/audio/audio_stream.h

```diff
@@ -1041,9 +1041,15 @@ static inline void audio_stream_fmt_conversion(enum ipc4_bit_depth depth,
 	*valid_fmt = (valid >> 3) - 2;
 
 	/* really 24_3LE */
-	if (valid == 24 && depth == 24) {
-		*frame_fmt = SOF_IPC_FRAME_S24_3LE;
-		*valid_fmt = SOF_IPC_FRAME_S24_3LE;
+	if (valid == 24) {
```

Comment:
U8 is adding also, be careful with U8 case for return at line 1052.
also I think we may need a switch to cover this function, too much if in this function.

may need rewrite this function after both merged, or later merged one to do this.

Reply:
thanks